EdgeOfEmpireBot/EdgeOfEmpireBot/MessageHandlers/GeneralMessageHandler/GeneralMessageHandler.cs

```diff
@@ -43,14 +45,31 @@ public async Task ProcessCommand(string userInput)
                 await messageService.SendMessage("[Sarcasm]: I dont remember...\n" +
                 $"[Statement]:That was a joke human.\n[Recalling]: I remember {remember}");
                 break;
-
+            case "changerpg":
+                await ChangeRpgs(commandParams[1]);
+                break;
             default:
                 // Not a custom command so try processing as a simple command.
                 await ProcessSimpleCommand(command);
                 break;
         }
     }
 
+    /// <summary>
+    /// Updates the commands that are shared across rpgs to the rpg that is provided
+    /// </summary>
+    private async Task ChangeRpgs(string rpg)
+    {
+        if (!ListOfRpgs.Contains(rpg))
+        {
+            await messageService.SendMessage("[Sad]: That RPG is not available...");
+            return;
+        }
+
+        await dataService.UpsertToJsonFile("Commands", ("roll", rpg));
```

Comment:
Not sure on this one,
As they're probably a handful of overlapping commands generally between RPGS, especially PBtA games like Sprawl, Apoc World, Dungeon World, Masks etc 
That said, this means we could have "Dungeon World" set as the RPGbut still call commands from "Sprawl" like "Act Under Pressure" instead of "Defy Danger" as one of those moves are a lot easier to read than the other, 

Though if we have custom varients of a game, ie we've got a handful of modified Sprawl moves, but maintained the default for reasons (mainly one offs with different people) which seems harder to implement this way? 

I might just have a think on this approach, might just be the case of having some additional JSON files then upserting that file into this, to maintain overalp as a whole

Reply:
This may need to be parked until the rpg games have been added. If there is as much overlap as you are suggesting and we are not able to abstract those out into generics then this will need to be architect-ed out prior to development. (Here comes the bot PBIs xD).

You and I will need to sit down at some point read through the different RPG rules design a solution/implementation plan for each of them (without yet developing them) then cross reference between all of them and see what parts may be plausible to generalise and which may not be possible

Reply:
Its worth noting that this is one of the pitfalls of having a general purpose bot. If we are going to be making RPGs and also customise those RPGs it still makes sense to update the process in the Commands.json. It would effectively be passing the baton of responsibility to one rpg at a time.

Being able to mix different methods/functions from different rpgs also just seems like a bad idea tbh. It makes the code way more error prone as you are tightly coupling the rpgs and giving them have too much responsibility.

Reply:
I feel like there's a good handful off overalp? Maybe there's not. 
Definetly a lot of overlap if we count custom varients of Sprawl (had a a few that would switch between when starting new campaigns), also some things like Help,Hinder between PBtA games. 
Not sure for other RPGs we might actually play? 
Might have a sus on what we might build up and then we can discuss if its actually gonna be a lot of overalp or I'm  just thinking there is